test/Microsoft.DotNet.Tools.Tests.Utilities/Assertions/DirectoryInfoAssertions.cs

```diff
@@ -1,94 +1,109 @@
-﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
-// Licensed under the MIT license. See LICENSE file in the project root for full license information.
-
-using FluentAssertions;
-using FluentAssertions.Execution;
-using Microsoft.DotNet.Cli.Utils;
 using System;
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
-using System.Threading.Tasks;
+using FluentAssertions;
+using FluentAssertions.Execution;
 
 namespace Microsoft.DotNet.Tools.Test.Utilities
 {
-    public class DirectoryInfoAssertions
+    public class DirectoryInfoAssertions : FileSystemInfoAssertions<DirectoryInfo, DirectoryInfoAssertions>
     {
-        private DirectoryInfo _dirInfo;
-
-        public DirectoryInfoAssertions(DirectoryInfo dir)
+        public DirectoryInfoAssertions(DirectoryInfo subject)
         {
-            _dirInfo = dir;
+            Subject = subject;
         }
 
-        public DirectoryInfo DirectoryInfo => _dirInfo;
-
-        public AndConstraint<DirectoryInfoAssertions> Exist()
+        public AndConstraint<DirectoryInfoAssertions> HaveFile(string expectedFile, string because = "", params object[] becauseArgs)
         {
-            Execute.Assertion.ForCondition(_dirInfo.Exists)
-                .FailWith("Expected directory {0} does not exist.", _dirInfo.FullName);
+            var immediateChildFiles = Subject.EnumerateFiles(expectedFile, SearchOption.TopDirectoryOnly);
+
+            Execute.Assertion
+                .ForCondition(immediateChildFiles.Any())
+                .BecauseOf(because, becauseArgs)
+                .FailWith($"Expected '{Subject.FullName}' to have file {expectedFile}{{reason}}, but it does not.");
+
             return new AndConstraint<DirectoryInfoAssertions>(this);
         }
 
-        public AndConstraint<DirectoryInfoAssertions> HaveFile(string expectedFile)
+        public AndConstraint<DirectoryInfoAssertions> NotHaveFile(string expectedFile, string because = "", params object[] becauseArgs)
         {
-            var file = _dirInfo.EnumerateFiles(expectedFile, SearchOption.TopDirectoryOnly).SingleOrDefault();
-            Execute.Assertion.ForCondition(file != null)
-                .FailWith("Expected File {0} cannot be found in directory {1}.", expectedFile, _dirInfo.FullName);
+            var immediateChildFiles = Subject.EnumerateFiles(expectedFile, SearchOption.TopDirectoryOnly);
+
+            Execute.Assertion
+                .ForCondition(!immediateChildFiles.Any())
```

Comment:
Good catch. I found another approach that is more expressive than both of these.